### ISSUE TITLE
Discussion: Behaviour when a gear produces invalid metadata

### ISSUE THREAD

**Issue by kofalt** [scitran/core#313]
We haven't explicitly laid out what should happen when a gear produces files but bad metadata. Right now, we reject the upload entirely. This is fine for gears whose runtime is measured in seconds, but we might have different opinions when runtime is measured in hours.

Personally, I would be very unhappy if 10 hours of runtime was thrown away due to a previously-undiscovered case where (for example) the gear produced an invalid timestamp. We ran into a similar scenario this morning (minus the 10 hours part) with the dicom classifier.

Possible options:
1. Keep the current behaviour - throw away the data. Gear authors need to be exceedingly careful.
2. Keep uploaded data, and discard the metadata.
3. Let the gear author decide. Proposed by Ryan, it might be that the data is worthless or invalid without the metadata. This could be exposed as a new manifest key, which the core would reference when making a decision.
4. Quarantine the data & metadata. Maybe save both as files to a "broken" acquisition and allow super-admins (or etc) to access. Or maybe just throw it on disk somewhere and log / record the incident.

Whatever the decision, we should make it explicitly and record the reasoning. I'm marking this as a bug, because right now a gear hitting this scenario will run 3 times, and that's clearly something we should fix.

Any changes blocked by EnginePlacer landing (ref #159).
Relevant to @lmperry, @ryansanford, @nagem. Assigning to @gsfr for team discussion.

**Comment by ryansanford** [comment]
Thanks for starting the discussion @kofalt 

Scope of option 3 is outside of scitran/core. 

Option 4 is attractive, but in practice will be a support/maintenance/complexity bourdon. I think this shou

**Comment by kofalt** [comment]
Agreed in general.

Option 3 will at least _involve_ the core, but indeed it would absolutely involve our other components.
It could definitely be an optional manifest key, in which case we could assume some default.

**Comment by gsfr** [comment]
I like the idea of option 3, including an optional manifest key, and defaulting to the current throw-away behavior. This, of course, assumes that metadata isn't invalid with respect to target location.

**Comment by kofalt** [comment]
Works for me pending other opinions. Since this would be a decent amount or work across components (and blocked anyway), I'm putting this in the icebox until we need it.

Changing to enhancement - the 3 retries component is tracked by #314.

**Comment by nagem** [comment]
Duplicate of #795.

**Comment by kofalt** [comment]
Closing as dupe
